### ISSUE TITLE
Get rid of "ln" (link) warnings in Console

### ISSUE THREAD

**Issue by Igazine** [Moonshine-IDE/Moonshine-IDE#1157]
_Note: This was originally an MSDKI issue, I recreated it in this repo as it belongs to Moonshine IDE._
_Original reference: https://github.com/Moonshine-IDE/Moonshine-SDK-Installer/issues/13_

Warnings that look like errors in the Console

```
: ln: /Users/-user-/Downloads/MoonshineSDKs/Haxe/haxe-4.2.4/libneko.dylib: File exists
: ln: /Users/-user-/Downloads/MoonshineSDKs/Haxe/haxe-4.2.4/libneko.2.dylib: File exists
: ln: /Users/-user-/Downloads/MoonshineSDKs/Haxe/haxe-4.2.4/libneko.2.3.0.dylib: File exists
: ln: /Users/-user-/Downloads/MoonshineSDKs/Haxe/haxe-4.2.4/std.ndll: File exists
: ln: /Users/-user-/Downloads/MoonshineSDKs/Haxe/haxe-4.2.4/mod_neko2.ndll: File exists
: ln: /Users/-user-/Downloads/MoonshineSDKs/Haxe/haxe-4.2.4/mysql.ndll: File exists
: ln: /Users/-user-/Downloads/MoonshineSDKs/Haxe/haxe-4.2.4/regexp.ndll: File exists
: ln: /Users/-user-/Downloads/MoonshineSDKs/Haxe/haxe-4.2.4/mysql5.ndll: File exists
: ln: /Users/-user-/Downloads/MoonshineSDKs/Haxe/haxe-4.2.4/zlib.ndll: File exists
: ln: /Users/-user-/Downloads/MoonshineSDKs/Haxe/haxe-4.2.4/sqlite.ndll: File exists
: ln: /Users/-user-/Downloads/MoonshineSDKs/Haxe/haxe-4.2.4/mod_tora2.ndll: File exists
: ln: /Users/-user-/Downloads/MoonshineSDKs/Haxe/haxe-4.2.4/ui.ndll: File exists
: ln: /Users/-user-/Downloads/MoonshineSDKs/Haxe/haxe-4.2.4/ssl.ndll: File exists
```

Issue: `ln` is being executed even if the links have been previously created.
[Related code](https://github.com/Moonshine-IDE/Moonshine-IDE/blob/master/ide/MoonshineDESKTOPevolved/src/actionScripts/utils/EnvironmentSetupUtils.as#L366)

**Comment by rat-moonshine** [comment]
My wild guess is the above problem/output starts showing since `OpenInTerminalPlugin` (#1038 ) introduced. It does some shell-reading during its activation but I'm not sure exactly what could side-effect out of it. 

The `activate(..)` from this plugin somehow affects the environment setup logic in Moonshine.. But if I changed the ordering of the plugin-intialization in `IFlexCoreBridgeImp`, then the problem resolves.

I see @Igazine already commit a fix/check to not appear the messages in console, meanwhile I also committed a fix https://github.com/Moonshine-IDE/Moonshine-IDE/commit/c27c4e12141190377f5424c6eafc2e2624ae5c46 from this side of the world. Hopefully, an additional fix won't hurt anyway.

**Comment by JoelProminic** [comment]
@Igazine reported that only `libneko.dylib` is checked for now - this assumes that either all of the links will exist, or none of them will exist.

I would prefer to check each of the links to avoid cases like this:
- One of the links failed or was interrupted on the original setup
- We add additional libraries that need links.

Otherwise, the updates look good.

**Comment by JoelProminic** [comment]
@Igazine reported that the `-F` option in `ln` will overwrite the existing links.   This may be a better solution, since it will work with updates.

**Comment by Igazine** [comment]
ln command has been changed to `ln -s -F -f`. According to ln man page:

>-F    If the target file already exists and is a directory, then remove it so that the link may occur.  The -F option should
           be used with either -f or -i options.  If neither -f nor -i is specified, -f is implied.  The -F option is a no-op
           unless -s is specified.

**Comment by JoelProminic** [comment]
Test | All links exist | Warnings on console
-- | -- | --
Remove all links and restart Moonshine | Yes | BUG: Yes
All links exist already | Yes | BUG:  Yes
Remove half of the links | Yes | BUG:  Yes
Remove the other half of the links | Yes | BUG:  Yes


I'm seeing these warnings again in all cases I tested:

    : ln: /Users/-user-/Downloads/MoonshineSDKs/Haxe/haxe-4.2.4/libneko.dylib: File exists
    : ln: /Users/-user-/Downloads/MoonshineSDKs/Haxe/haxe-4.2.4/libneko.2.dylib: File exists
    : ln: /Users/-user-/Downloads/MoonshineSDKs/Haxe/haxe-4.2.4/libneko.2.3.0.dylib: File exists
    : ln: /Users/-user-/Downloads/MoonshineSDKs/Haxe/haxe-4.2.4/std.ndll: File exists
    : ln: /Users/-user-/Downloads/MoonshineSDKs/Haxe/haxe-4.2.4/mod_neko2.ndll: File exists
    : ln: /Users/-user-/Downloads/MoonshineSDKs/Haxe/haxe-4.2.4/mysql.ndll: File exists
    : ln: /Users/-user-/Downloads/MoonshineSDKs/Haxe/haxe-4.2.4/regexp.ndll: File exists
    : ln: /Users/-user-/Downloads/MoonshineSDKs/Haxe/haxe-4.2.4/mysql5.ndll: File exists
    : ln: /Users/-user-/Downloads/MoonshineSDKs/Haxe/haxe-4.2.4/zlib.ndll: File exists
    : ln: /Users/-user-/Downloads/MoonshineSDKs/Haxe/haxe-4.2.4/sqlite.ndll: File exists
    : ln: /Users/-user-/Downloads/MoonshineSDKs/Haxe/haxe-4.2.4/mod_tora2.ndll: File exists
    : ln: /Users/-user-/Downloads/MoonshineSDKs/Haxe/haxe-4.2.4/ui.ndll: File exists
    : ln: /Users/-user-/Downloads/MoonshineSDKs/Haxe/haxe-4.2.4/ssl.ndll: File exists


Here is my test command:

    cd ~/Downloads/MoonshineSDKs/Haxe/haxe-4.2.4
    ls libneko.dylib libneko.2.dylib libneko.2.3.0.dylib std.ndll mod_neko2.ndll mysql.ndll regexp.ndll mysql5.ndll zlib.ndll sqlite.ndll mod_tora2.ndll ui.ndll ssl.ndll; echo $?

**Comment by Igazine** [comment]
@JoelProminic  Is it possible that you (or the build server) did not pull the latest changes from MSDKI?
The [code part that runs the ln commands](https://github.com/Moonshine-IDE/Moonshine-SDK-Installer/commit/7c1afd77cd0321d338719eb1387b8f506696b5e8) is in MSDKI's code base

**Comment by JoelProminic** [comment]
Ah, I see the build was stuck on an old branch of MSDKI.   I started a new build, and I'll plan to do a test when it finishes

I'm still surprised that I saw the full list of warnings on every run, though, even when the links had been removed.  Could this action be triggered in two different locations?

**Comment by Igazine** [comment]
The previous "fix" ignored this symlink command if at least 1 link already existed, and this fix was added to Moonshine-IDE. The new fix forces the creation of the symlinks even if they already exist, so warnings should not be displayed anymore. But since this array of commands is in MSDKI/InstallerSharedCore, it has to be updated (pulled, fetched) from MSDKI's repo.

It's possible that EnvironmentSetupUtil.getPlatformCommand() (or the caller functions) are being referenced multiple times in the code, but in that case you'd see all 13 warnings multiple times.

Maybe this logic is a bit dated? Symlinks are only needed if MSDKI or Moonshine installs Haxe and Neko. Maybe both can take care of the creation of the symlinks only once after the installation finishes?

**Comment by JoelProminic** [comment]
Testing with the master branch of MSDKI, this looks good.  This this update it doesn't really matter if Moonshine running the commands more than once, since there is no warning.

Test | All links exist | Warnings on console
-- | -- | --
Remove all links and restart Moonshine | Yes | No
All links exist already | Yes | No
Remove half of the links | Yes | No
Remove the other half of the links | Yes | No

This is ready for a merge.

**Comment by Igazine** [comment]
The branch has been merged into master

**Comment by JoelProminic** [comment]
I confirmed this in master.
